debezium-connector-mongodb/src/main/java/io/debezium/connector/mongodb/metrics/MongoDbStreamingChangeEventSourceMetrics.java

```diff
@@ -44,6 +57,47 @@ public long getNumberOfDisconnects() {
         return numberOfDisconnects.get();
     }
 
+    @Override
+    public long getLastSourceEventPollTime() {
+        return lastSourceEventPollTime.get();
+    }
+
+    @Override
+    public long getLastEmptyPollTime() {
+        return lastEmptyPollTime.get();
+    }
+
+    @Override
+    public long getNumberOfSourceEvents() {
+        return numberOfSourceEvents.get();
+    }
+
+    @Override
+    public long getNumberOfEmptyPolls() {
+        return numberOfEmptyPolls.get();
+    }
+
+    public void onSourceEventPolled(ChangeStreamDocument<BsonDocument> event, Clock clock, Instant prePollTimestamp) {
+        AtomicLong counter;
+        AtomicLong timer;
+
+        if (event == null) {
+            counter = numberOfEmptyPolls;
+            timer = lastEmptyPollTime;
+        }
+        else {
+            counter = numberOfSourceEvents;
+            timer = lastSourceEventPollTime;
+        }
+
+        counter.incrementAndGet();
+        if (clock != null && prePollTimestamp != null) {
+            var now = clock.currentTimeAsInstant();
```

Comment:
@jcechace This should be the first statement in this method.
When can `clock` be empty?

Reply:
True.. When you don't provide them :) If you think that the guarding is pointless here I can remove it.

Reply:
Updated. It should be OK now.